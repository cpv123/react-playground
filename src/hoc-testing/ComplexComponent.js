import React from 'react';
import PropTypes from 'prop-types';
import { Button, Typography, withStyles  } from '@material-ui/core';
import { compose, withState, withHandlers, onlyUpdateForKeys } from 'recompose';

const styles = {
    root: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
};

export function ComplexComponent({ 
    classes, 
    heading, 
    incrementButtonTitle, 
    buttonTitle 
}) {
    return (
        <div className={classes.root}>
            <Typography>{heading}</Typography>
            <Button
                variant="contained"
                color="primary"
                onClick={incrementButtonTitle}
            >
                {buttonTitle}
            </Button>
        </div>
    );
}

ComplexComponent.propTypes = {
    classes: PropTypes.object.isRequired,
    heading: PropTypes.string.isRequired,
    incrementButtonTitle: PropTypes.func.isRequired,
    buttonTitle: PropTypes.number.isRequired,
};

const ComplexComponentWithHocs = compose(
    withStyles(styles),
    withState('buttonTitle', 'setButtonTitle', 0),
    withHandlers({
        incrementButtonTitle: ({ buttonTitle, setButtonTitle }) => () => {
            setButtonTitle(buttonTitle => buttonTitle + 1)
        }
    }),
    onlyUpdateForKeys([ "buttonTitle" ])
)(ComplexComponent);

export default ComplexComponentWithHocs;