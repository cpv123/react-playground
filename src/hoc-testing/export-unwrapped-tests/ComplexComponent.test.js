import React from 'react';
import { shallow } from 'enzyme';
// Import the unwrapped component this time i.e. not the default export
import { ComplexComponent } from '../ComplexComponent';

describe('the unwrapped ComplexComponent', () => {

    it('renders as expected', () => {
        // We must provide all props, as we've imported the base component only
        const mockProps = {
            classes: {},
            heading: "Test heading",
            incrementButtonTitle: jest.fn(),
            buttonTitle: 0,
        };
        const wrapper = shallow(<ComplexComponent {...mockProps} />);
        expect(wrapper).toMatchSnapshot()
    });

});