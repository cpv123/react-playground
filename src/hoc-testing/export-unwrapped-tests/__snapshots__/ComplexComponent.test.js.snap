// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the unwrapped ComplexComponent renders as expected 1`] = `
<div>
  <WithStyles(Typography)>
    Test heading
  </WithStyles(Typography)>
  <WithStyles(Button)
    color="primary"
    onClick={[MockFunction]}
    variant="contained"
  >
    0
  </WithStyles(Button)>
</div>
`;
